platforms/fabric/src/main/java/mod/chiselsandbits/fabric/platform/configuration/FabricConfigurationBuilder.java

```diff
@@ -72,6 +73,34 @@ public <T> Supplier<List<? extends T>> defineList(final String key, final List<T
         return value;
     }
 
+    @Override
+    public Supplier<Vector4f> defineVector4f(final String key, final Vector4f defaultValue)
+    {
+        final FabricConfigurationValue<Vector4f> value = new FabricConfigurationValue<>(
+                source,
+                key,
+                jsonElement -> {
+                    if (!jsonElement.isJsonArray())
+                        throw new JsonParseException("List: " + key + " is not an array");
+
+                    final JsonArray jsonArray = jsonElement.getAsJsonArray();
+                    final List<Float> list = new ArrayList<>();
+
+                    for (final JsonElement element : jsonArray) {
+                        list.add(GSON.fromJson(element, Float.class));
+                    }
+
+                    return new Vector4f(list.get(0), list.get(1), list.get(2), list.get(3));
```

Comment:
Please validate the length so that it is either 3 or 4, when it is 3 make the last value 1